### PR TITLE
add function to process mosaic based on WarpedVRT

### PR DESCRIPTION
**Pull request by justinkotzebue**

### CODE REVIEW

File: gis_utils/r/mosaic.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    infiles = _open_reprojected_vrts(infiles, dst_crs, dst_nodata=dst_nodata,
+                                     src_nodata=src_nodata)
+
+    out_image, out_trans = merge(infiles, res=res, nodata=dst_nodata)
+    if out_meta is None:
+        out_meta = infiles[0].meta.copy()
+
+
+    _crs = rasterio.crs.CRS({'init': dst_crs})
```
Comment (j08lue) [inline gis_utils/r/mosaic.py]: no need to use a private variable here (`_crs`). You are in a function. Just call it `crs`.

Comment (j08lue) [reply gis_utils/r/mosaic.py]: You should not convert `dst_crs` from some string to a dict inside the function. Leave that to the user / script that calls the function. You are adding a level of abstraction (your own input format) that is not necessary when the `rasterio` standard (`dict` or `rasterio.crs.CRS`) is easy enough.

Stuff like this belongs into a much higher-level **interface**.

File: gis_utils/r/__init__.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
 from . import clip
 from . import resample_array
 from . import sample_values
+from . import mosaic
```
Comment (j08lue) [inline gis_utils/r/__init__.py]: I am really not a big fan of this `__init__.py` that imports **all submodules**. I know why you are doing that: so you can `from gis_utils import r` and then `r.mosaic.some_function`. But I don't like that either, to be honest. :smile:

If you take `rasterio` as an example, you have to `import rasterio.warp` in order to be able to call `rasterio.warp.reproject`. So it is not enough to `import rasterio`. There is a reason for this slight inconvenience: You actually only import what you need, not the entire package. 

If large packages like `rasterio` or `scipy` required us to import the whole package every time we need one function, our scripts would have a very long load time.

The current practice in `gis_utils` means we practically import all of `rasterio` that is used anywhere in the package every time we just want to use one of the functions. I think you should rather get in the habit of `import gis_utils.r.mosaic` and then `gis_utils.r.mosaic.mosaic`. It is not pretty, but adds **clarity and efficiency**.

https://google.github.io/styleguide/pyguide.html#Imports

Comment (justinkotzebue) [reply gis_utils/r/__init__.py]: I see your point Jonas, lets just do it then in order to have a more efficient code!

**Pull request merged**